### DISCUSSION TITLE
Show row selection checkbox

### DISCUSSION THREAD

**Discussion by visshaljagtap** [Comcast/react-data-grid#3045] [Q&A]
I am using version 7.0.0-beta.12

How to show checkbox selection for rows?

I have did the following configuration

<DataGrid
        headerRowHeight={filters?.enabled ? 70 : undefined}
        style={{ height: '95%' }}
        className='rdg-light fill-grid'
        columns={columns}
        rows={data ? data : []}
        rowKeyGetter={rowKeyGetter}
        selectedRows={selectedRows}
        onSelectedRowsChange={setSelectedRows}
        defaultColumnOptions={{
          sortable: true,
          resizable: true
        }}
      />

**Comment by visshaljagtap** [comment]
Also renderers props is not working

**Comment by paustint** [comment]
@visshaljagtap - Did you make sure to add a column for the checkbox?

Look for `SelectColumn,` in the examples https://github.com/adazzle/react-data-grid/blob/main/website/demos/AllFeatures.tsx

**Comment by rohit-squirrel** [comment]
Hi, i am using the SelectColumn variable for the checkbox, but having row selection issue i.e when i click on any row it is selecting all the rows. When i checked for the selectedRow it gave me the row which i clicked on and an undefined value(i think which is for select-all check box), which is trigerring the all row selection.
Can you help me with this issue? @paustint @amanmahajan7

**Reply by paustint** [reply]
Hard to say exactly, but I suspect that you are not properly setting the key on each of your rows and you have rows that have the same key (potentially a key of `undefined`) here are the places I would check:
1. Make 100% sure all your rows have a unique key and use `getRowKey` (debug the function you provide here to be sure you are returning what you think you are)
2. Debug the state you have in here `const [selectedRows, setSelectedRows] = useState(() => new Set());` and see what values are included in `selectedRows` and compare that to `#1`
